test/ably/realtime/realtimechannel_test.py

```diff
@@ -468,3 +468,69 @@ async def test_channel_error_cleared_upon_connect_from_terminal_state(self):
         assert channel.error_reason is None
 
         await ably.close()
+
+    async def test_channel_params_received_by_relatime(self):
+        ably = await TestApp.get_ably_realtime()
+        channel_name = random_string(5)
+        channel = ably.channels.get(channel_name, ChannelOptions(params={
+            "rewind": "1"
+        }))
+        await channel.attach()
+        assert channel.params["rewind"] == "1"
+
+        await ably.close()
+
+    async def test_channel_params_unknown_params_skipped_by_relatime(self):
+        ably = await TestApp.get_ably_realtime()
+        channel_name = random_string(5)
+        channel = ably.channels.get(channel_name, ChannelOptions(params={
+            "rewind": "1",
+            "foo": "bar"
+        }))
+        await channel.attach()
+        assert channel.params["rewind"] == "1"
+        assert channel.params.get("foo") is None
+
+        await ably.close()
+
+    async def test_channel_params_as_dict(self):
+        ably = await TestApp.get_ably_realtime()
+        channel_name = random_string(5)
+        channel = ably.channels.get(channel_name, {
+            "params": {"delta": "vcdiff"}
+        })
+        await channel.attach()
+        assert channel.params["delta"] == "vcdiff"
+
+        await ably.close()
+
+    async def test_channel_get_channel_with_same_params(self):
+        ably = await TestApp.get_ably_realtime()
+        channel_name = random_string(5)
+        channel = ably.channels.get(channel_name, {
+            "params": {"rewind": "1"}
+        })
+        await channel.attach()
+        same_channel = ably.channels.get(channel_name, {
+            "params": {"rewind": "1"}
+        })
+        assert channel == same_channel
+
+        await ably.close()
+
+    async def test_channel_get_channel_with_different_params(self):
+        ably = await TestApp.get_ably_realtime()
+        channel_name = random_string(5)
+        channel = ably.channels.get(channel_name, {
+            "params": {"rewind": "1"}
+        })
+        await channel.attach()
+
+        with pytest.raises(AblyException):
```

Comment:
@coderabbitai can we assert on exception contents here? I want to check for `statusCode` nd `code` with message ( optional )

Reply:
This can also be applied across the codebase wherever an exception is expected.